components/accountPreferences/EmailPreferences.tsx

```diff
@@ -0,0 +1,108 @@
+import React from "react";
+import { useRequestEmailChangeMutation } from "@/saleor/api";
+import { useForm } from "react-hook-form";
+interface EmailChangeFormData {
+  newEmail: string;
+  password: string;
+  redirectUrl: string;
+}
+
+export const EmailPreferences: React.VFC<any> = ({}) => {
+  const [requestEmailChange] = useRequestEmailChangeMutation({});
+  const [successMessage, setSuccessMessage] = React.useState<String | null>(
```

Comment:
Can we relay here on the type inference to simplify this code? Is the `null` declaration necessary?